locale.c

```diff
@@ -1730,6 +1725,16 @@ S_calculate_LC_ALL_string(pTHX_ const char ** category_locales_list)
      * categories, adding new ones as they show up on obscure platforms.
      */
 
+    const char * my_category_locales_list[LOCALE_CATEGORIES_COUNT_];
+    const char ** locales_list = category_locales_list;
+    if (locales_list == NULL) {
+        locales_list = (const char **) &my_category_locales_list;
```

Comment:
This should just be:
```
        locales_list = my_category_locales_list;
```
An array of T  (T = `const char *` here) decays to a pointer to T in most cases in C (`const char **` in this case).

The `&` on the array name is one of the few exceptions\*, which would make `&my_category_locales_list` a pointer to an array of LOCALE_CATEGORIES_COUNT_ pointers to const char *, which I suspect led you to adding the cast.

\* the other one I can think of is sizeof().